js/client-chat.js

```diff
@@ -641,8 +658,31 @@
 								buffer += '</td><td>'+row.w+'</td><td>'+row.l+'</td><td>'+row.t+'</td></tr>';
 							}
 						}
+						if (hiddenFormats.length) {
+							if (hiddenFormats.length === formatLength) {
+								buffer += '<tr class="no-matches"><td colspan="8"><em>This user has not played any ladder games that match the format targeting.</em></td></tr>';
+							}
+
+							buffer += '<tr><td colspan="8"><button class="show-other-formats">' + hiddenFormats.slice(0, 3).join(', ') + (hiddenFormats.length > 3 ? ' and ' + (hiddenFormats.length - 3) + ' other formats' : '') + ' not shown</button></td></tr>';
+						}
 						buffer += '</table></div>';
 						self.add('|raw|'+buffer);
+
+						if (hiddenFormats.length) {
+							self.$chatFrame.find('button.show-other-formats:last').click(function (e) {
```

Comment:
You don't even need another event listener; presumably our `button name` system should be enough.